### PR TITLE
new(modern_bpf): add support for `set` family syscalls

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-modern-bpf

/area libpman

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR is part of a series https://github.com/falcosecurity/libs/issues/513, the final aim is to support the most important syscalls also in the new probe. This PR introduces:

* `setgid`
* `setuid`
* `setns`
* `setpgid`
* `setresgid`
* `setresuid`
* `setsid`
* `setrlimit`
* `prlimit64`

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(modern_bpf): add support for `set` family syscalls
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]
:heart:

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	if(!ringbuf__reserve_space(&ringbuf, SETPGID_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_SETPGID_E, SETPGID_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: pid (type: PT_FD) */
```
Comment (hbrueckner) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c:27]: Is PT_FD here correcot, shouldn't it be PT_PID?

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	if(!ringbuf__reserve_space(&ringbuf, SETPGID_X_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_SETPGID_X, SETPGID_X_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_PID)*/
```
Comment (hbrueckner) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c:61]: PT_ERRNO?

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setresuid.bpf.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	if(!ringbuf__reserve_space(&ringbuf, SETRESUID_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_SETRESUID_E, SETRESUID_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: ruid (type: PT_GID) */
```
Comment (hbrueckner) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setresuid.bpf.c:27]: PT_UID instead of PT_GID?

**Comment by hbrueckner** [comment]
@Andreagit97  Other than those remarks above, your PR lgtm. Thanks.

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c (hunk 1 above)
Comment (Andreagit97) [reply driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c:27]: thank you

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c (hunk 2 above)
Comment (Andreagit97) [reply driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setpgid.bpf.c:61]: In the event table, we have used `PT_PID`  https://github.com/falcosecurity/libs/blob/2fa7b0dedd0c87d53fb46c5e264b94bdf47b8d19/driver/event_table.c#L310 so right now i will keep it like it is, but yeah we need to find a good convention! probably we will work on that here https://github.com/falcosecurity/libs/pull/526

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setresuid.bpf.c (hunk 3 above)
Comment (Andreagit97) [reply driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/setresuid.bpf.c:27]: yep, thank you!

**Comment by Andreagit97** [comment]
I will open a final PR to clean up all this stuff, I've added your comments to my agenda :) Thank you for the review

**Pull request merged**